### PR TITLE
Fix a useless error message with something more useful

### PR DESCRIPTION
**Pull request by buggtb**
Currently if the ingestion fails with no product type you get the message "null" because the server throws an npe and the client gets nothing in return. I've prevented the NPE and returned some more useful information

### CODE REVIEW

**Comment by chrismattmann** [comment]
Big +1! @buggtb

File: filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManager.java [hunk 1]
```diff
@@ -18,30 +18,16 @@
...
-import org.apache.oodt.cas.filemgr.structs.Product;
-import org.apache.oodt.cas.filemgr.structs.Query;
-import org.apache.oodt.cas.filemgr.structs.Reference;
-import org.apache.oodt.cas.filemgr.structs.exceptions.CatalogException;
-import org.apache.oodt.cas.filemgr.structs.exceptions.QueryFormulationException;
-import org.apache.oodt.cas.filemgr.structs.exceptions.RepositoryManagerException;
-import org.apache.oodt.cas.filemgr.structs.exceptions.DataTransferException;
-import org.apache.oodt.cas.filemgr.structs.exceptions.ValidationLayerException;
-import org.apache.oodt.cas.filemgr.structs.exceptions.VersioningException;
+import org.apache.oodt.cas.filemgr.structs.*;
```
Comment (chrismattmann) [inline filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManager.java]: please don't use * imports. Please make sure it specifically imports classes.

File: filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManager.java [hunk 2]
```diff
@@ -18,30 +18,16 @@
...
+import org.apache.oodt.cas.filemgr.structs.exceptions.*;
```
Comment (chrismattmann) [inline filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManager.java]: no star imports

File: filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java [hunk 3]
```diff
@@ -18,48 +18,34 @@
...
-import org.apache.oodt.cas.filemgr.structs.Query;
-import org.apache.oodt.cas.filemgr.structs.Reference;
-import org.apache.oodt.cas.filemgr.util.GenericFileManagerObjectFactory;
-import org.apache.oodt.cas.filemgr.util.XmlRpcStructFactory;
-import org.apache.oodt.cas.filemgr.versioning.Versioner;
-import org.apache.oodt.cas.filemgr.structs.exceptions.VersioningException;
 import org.apache.oodt.cas.filemgr.datatransfer.DataTransfer;
-import org.apache.oodt.cas.filemgr.structs.exceptions.DataTransferException;
-import org.apache.oodt.cas.filemgr.structs.exceptions.ConnectionException;
+import org.apache.oodt.cas.filemgr.structs.*;
```
Comment (chrismattmann) [inline filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java]: no star imports

File: filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java [hunk 4]
```diff
@@ -18,48 +18,34 @@
...
+import org.apache.oodt.cas.filemgr.structs.exceptions.*;
```
Comment (chrismattmann) [inline filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java]: no star imports

File: filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java [hunk 5]
```diff
@@ -1286,10 +1272,24 @@ public String ingestProduct(Product product, Metadata metadata,
...
+            argList.add(productHash);
+            client.execute("filemgr.removeProduct", argList);
+          } catch (Exception e1) {
+            LOG.log(Level.SEVERE, "Failed to rollback ingest of product ["
+                                  + product + "] : " + e2.getMessage());
+          }
+          throw new Exception(e2);
+        }
+        catch (Exception e) {
+            //e.printStackTrace();
```
Comment (chrismattmann) [inline filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java]: remove out commented code if it's not part of the patch.

File: filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java (hunk 3 above)
Comment (buggtb) [inline filemgr/src/main/java/org/apache/oodt/cas/filemgr/system/XmlRpcFileManagerClient.java]: Boo, intellij rolls them up automatically over a certain amount, I'll find the setting.

**Comment by buggtb** [comment]
Check again please, Professor Mattmann....

**Comment by MJJoyce** [comment]
:+1: (minus the merge conflicts =( )

**Comment by buggtb** [comment]
Aye i'll unmunge it
